### PR TITLE
Cleanups and small fixes

### PR DESCRIPTION
**Pull request by maage**
I guess most of these fixes can be picked individally too.

Most are just cleanups where I try to modify code to match my view of how things should be.

Fixes are bugfixes.

Changes modify current functionality and can break something. Or give error where there were none before.

### CODE REVIEW

**Comment by jedisct1** [comment]
Wow, this is amazing!

Thanks a ton for having taken the time to review that code and add all these improvements!

This is my first project in Go, so these changes are very helpful.

Thanks again!

File: dnscrypt-proxy/proxy.go [hunk 1]
```diff
@@ -13,6 +13,7 @@ import (
 	"github.com/jedisct1/dlog"
 	clocksmith "github.com/jedisct1/go-clocksmith"
 	stamps "github.com/jedisct1/go-dnsstamps"
+	"github.com/miekg/dns"
```
Comment (dapphp) [inline dnscrypt-proxy/proxy.go:16]: I know nothing about Go, but am wondering if pulling in a fairly large library for one rcode is worth it or has any negative side effects (increase binary size).

Comment (jedisct1) [reply dnscrypt-proxy/proxy.go:16]: It's already used for many more things in other files :)

**Comment by jedisct1** [comment]
Honestly one of the best PR I've ever received.

**Pull request merged**